### DISCUSSION TITLE
Document how can we create the Windows boot images

### DISCUSSION THREAD

**Discussion by rgl** [ipxe/wimboot#9] [General]
@mcb30, Can you please document how we can create the Windows boot images that were introduced at https://github.com/ipxe/wimboot/commit/38c3c87cf1cf752ef7bb7ebe53e85ace1743752e?

PS I'm really linking seeing how you are creating the qa tests :-)

**Comment by mcb30** [comment]
> PS I'm really linking seeing how you are creating the qa tests :-)

Thank you!

> @mcb30, Can you please document how we can create the Windows boot images that were introduced at [38c3c87](https://github.com/ipxe/wimboot/commit/38c3c87cf1cf752ef7bb7ebe53e85ace1743752e)?

I've just added an automated tests [`README.md`](https://github.com/ipxe/wimboot/blob/master/test/README.md) - I hope that explains what you need.

**Comment by rgl** [comment]
Thank you!

I almost got it working in my Ubuntu 20.04 machine with the minor changes at https://github.com/ipxe/wimboot/pull/8.

With the windows 2019 trial iso from `https://software-download.microsoft.com/download/pr/17763.737.190906-2324.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us_1.iso` (which I'm using at https://github.com/rgl/windows-vagrant/blob/master/windows-2019.json#L4) windows refuses to boot with:

![image](https://user-images.githubusercontent.com/43356/107616089-415ba380-6c45-11eb-8331-6f75c349d3d0.png)

Can you try it out?

**Comment by mcb30** [comment]
> I almost got it working in my Ubuntu 20.04 machine with the minor changes at #8.

Now merged; thank you!

> With the windows 2019 trial iso from https://software-download.microsoft.com/download/pr/17763.737.190906-2324.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us_1.iso

I downloaded WIndows Server 2019 from this link and it works for me with no problems.  I have added it to the test matrix in commit https://github.com/ipxe/wimboot/commit/eb8ef32.

You can see a successful test run on Ubuntu 20.04 including the Windows Server 2019 test image at https://github.com/ipxe/wimboot/runs/1878806551.  The test logs include the output from `./testwimboot -vv` which shows the full VM definitions as created: you might want to compare these to the output you get on your local system.

I note that your screenshot shows the title for the `Windows 10` test case - is it possible that you've somehow mixed up your test files?

**Reply by rgl** [reply]
With eb8ef323d2b5f9878c5d77d2fc99ccc677eae787 it does not work.

But since 699f81059fe879b44e8de3a4e1a12dbd0c88d01d (2.7.0) it works! :-)

I failed to mention that I didn't compile wimboot so I was using the version that was in the repo.

Thank you for this!
